Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Ride.hs

```diff
@@ -109,7 +112,47 @@ buildRideListItem QRide.RideItem {..} = do
       }
 
 ---------------------------------------------------------------------------------------------------
+ticketRideList :: ShortId DM.Merchant -> Maybe (ShortId Common.Ride) -> Maybe Text -> Maybe Text -> Maybe Text -> Flow Common.TicketRideListRes
+ticketRideList merchantShortId mbRideShortId countryCode mbPhoneNumber _ = do
+  merchant <- findMerchantByShortId merchantShortId
+  let totalRides = 5
+  let mbShortId = coerce @(ShortId Common.Ride) @(ShortId DRide.Ride) <$> mbRideShortId
+  case mbShortId of
+    Just shortId -> do
+      ride <- QRide.findRideByRideShortId shortId >>= fromMaybeM (InvalidRequest "Ride ShortId Not Found")
+      rideDetail <- rideInfo merchantShortId (cast ride.id)
+      let ticketRideDetail = makeRequiredRideDetail ride.driverId (ride, rideDetail)
+      return Common.TicketRideListRes {rides = [ticketRideDetail]}
+    Nothing -> case mbPhoneNumber of
+      Just number -> do
+        no <- getDbHash number
+        let code = fromMaybe "+91" countryCode
+        person <- runInReplica $ QPerson.findByMobileNumberAndMerchant code no merchant.id >>= fromMaybeM (PersonWithPhoneNotFound number)
+        ridesAndBooking <- QRide.findAllByDriverId person.id (Just totalRides) (Just 0) Nothing Nothing Nothing
```

Comment:
why do you fetch `ridesAndBooking` if use only rides?

Reply:
because I required a function which returns rides with given limit and offset , in place of creating something new , I reused it